### PR TITLE
Bugfix/acna 3518 clean build

### PR DESCRIPTION
**Pull request by dthampy**
## Description

Implement clean command `aio app clean-build` to remove build artifacts from the local machine without affecting deployed resources

## Related Issue

[ACNA-3518](https://jira.corp.adobe.com/browse/ACNA-3518) 

## Motivation and Context

The `aio app clean` command is critical for developers who need to get a clean build environment without manually finding and deleting build files. During development, stale build artifacts can cause inconsistent behavior, especially when switching between different environments or configurations. 

This command provides a reliable way to clean:
- Action build artifacts
- Web assets (both production and development)
- Distribution directories while preserving deployment tracking file
- Build tracking files

## How Has This Been Tested?

testing was performed for each command flag as described in the README.md:

- Tested `aio app clean-build` with default options to verify action and web assets are cleaned
- Tested `aio app clean-build --no-actions` to verify only web assets are cleaned
- Tested `aio app clean-build --no-web-assets` to verify only actions are cleaned
- Tested `aio app clean-build --dist-dir` to verify the entire dist directory is cleaned while preserving deployment tracking
- Tested `aio app clean-build --tracking-files` to verify build tracking files are cleaned
- Tested `aio app clean-build --dev` to verify only development web assets are cleaned
- Tested `aio app clean-build --prod` to verify only production web assets are cleaned
- Verified integration with other commands by testing the sequence:

  - `aio app build`
  - `aio app deploy`
  - `aio app clean-build --no-tracking-files`
  - `aio app run`

Additionally, created dedicated tests for edge cases including:
- When parent directory references are missing for web assets
- Error handling when attempting to restore deployment tracking files
- Various error formats to ensure proper error propagation

All tests pass with 100% branch coverage 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/commands/app/clean-build.js [hunk 1]
```diff
@@ -0,0 +1,293 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
```
Comment (moritzraho) [inline src/commands/app/clean-build.js:2]: nit 2025

File: package.json [hunk 2]
```diff
@@ -67,6 +67,7 @@
     "eslint-plugin-n": "^15.7.0",
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-promise": "^6.6.0",
+    "istanbul": "^0.4.5",
```
[the remark this answers is not in the document]
Comment (dthampy) [reply package.json:70]: Oh yeah I will remove that. 
I was consistently running into jest coverage issue for few lines/branches although I thought I added tests for them. So installed to help me identify untested parts of the code. Thanks for pointing that out

File: test/commands/app/clean-build.test.js [hunk 3]
```diff
@@ -0,0 +1,1511 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
```
[the remark this answers is not in the document]
Comment (dthampy) [reply test/commands/app/clean-build.test.js:2]: will update it

**Comment by dthampy** [comment]
Closing this PR in favor of [858](https://github.com/adobe/aio-cli-plugin-app/pull/858).
After a discussion with @purplecabbage, we agreed that it’s simpler and more consistent to always clean all build artifacts, rather than exposing individual flags. A
ll of the consolidated changes have been merged into [PR #858](https://github.com/adobe/aio-cli-plugin-app/pull/858)—please take a look over there.
